test/test_torch.py

```diff
@@ -17224,6 +17224,76 @@ def test_reciprocal_complex_extremal(self, device, dtype):
 
         self.compare_with_numpy(torch.reciprocal, np.reciprocal, vals, device, dtype)
 
+    def test_copysign_bool(self, device):
+        # Promoted to float
+        # Only 0 and 1
+        x = torch.tensor([0, 0, 1, 1], device=device, dtype=torch.bool)
+        y = torch.tensor([0, 1, 0, 1], device=device, dtype=torch.bool)
+        res = torch.tensor([0, 0, 1, 1], device=device, dtype=torch.float)
+        self.assertEqual(torch.copysign(x, y), res)
+
+    def _test_copysign(self, device, dtype, result_dtype):
+        # Scalar
+        x = torch.tensor([-1, 0, -0, 1], dtype=dtype)
```

Comment:
-0 is actually just 0. Only floating point allows for a signed zero; the representation of integers as two's complement doesn't allow for a signed zero. If we want to test behavior on negative zero we should use 0.0 and -0.0 instead of 0 and -0.